packages/base-styles/_mixins.scss

```diff
@@ -117,11 +117,52 @@
  * Focus styles.
  */
 
-@mixin block-toolbar-button-style__focus() {
-	box-shadow: inset 0 0 0 $border-width-focus var(--wp-admin-theme-color), inset 0 0 0 4px $white;
+@mixin focus-shadow(
```

Comment:
This mixin feels too complex to me. I understand you're trying to create a single central place to handle focus everywhere, but I don't think we can or should do this, as there will always be edgecases where you need to create a unique focus style just for the particular case. For example, the windows 10 outline should probably remain explicit CSS, even if it's duplicated across the codebase. This is a point Riad has driven home to me many times, when I got a bit too creative with variables and mixins — it reduces readability of the code.

Looping back to the problem at hand, 1.5px doesn't work well in non retina screens. And we see rules like the following a lot:

```
box-shadow: 0 0 0 $border-width-focus var(--wp-admin-theme-color);
```

What we want, in order to address this, if I'm not mistaken, is something like this instead, correct?

```
box-shadow: 0 0 0 $border-width-focus var(--wp-admin-theme-color);

@media ( min-resolution: 96dpi ) {
	box-shadow: 0 0 0 $border-width-focus-2x var(--wp-admin-theme-color);
}
```

Is there a simpler way we can use sass to simplify this? The emphasis would be to keep the code readable for others.